src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,160 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.HelpCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+import duke.exception.CommandException;
+
+/**
+ * Parser class to parse the input from users and return the appropriate instance of a command
+ */
+public class Parser {
+
+    /**
+     * Constructor for a Parser
+     */
+    public Parser() {
+    }
+
+    /**
+     * Parses the line input by the user
+     * @param line the line input by the uer
+     * @return A command corresponding to the input
+     * @throws CommandException when command is not understood
+     * @throws IndexOutOfBoundsException when command delete is called on an index that is out of bounds
+     * @throws NumberFormatException when command delete is called on an index that is not a number
+     */
+    public Command parse(String line) throws CommandException, IndexOutOfBoundsException, NumberFormatException {
+        String command = line.split(" ")[0];
+        try {
+            switch (command) {
```

Comment:
the switch statement as provided in the coding standard has no curly braces, and an explicit "//Fallthrough" comment should be included whenever there is a case statement without a break statement. it's a minor matter but perhaps you could add this in :)